Automattic/wp-calypso
# Enable Plugins pages for logged-out users

#### Proposed Changes
More info: pau2Xa-4j7-p2

This allows logged-out users to visit the following Plugin pages:
* The /plugins landing page
* The plugins detail pages – eg. /plugins/mailpoet
* The plugins category pages – eg. /plugins/browse/booking
* Plugins search result pages – eg. /plugins?s=commerce & /plugins/browse/design?s=ecommerce

Changes for this include:
* adjusting the section config to support logged-out routes
* adds data-less SSR routes
* enable localization for logged-out instances by using the URL param (ex. /es/plugins/mailpoet)
* hide current site-specific sections from logged-out users



#### Testing Instructions
* If you're running this on a locale instance, you'll need to run `yarn build-languages` so that the translations work
* Make sure you're logged out
* Go to `/plugins`
* The search box, categories links, and plugins should load and work properly
<img width="320" alt="Screenshot on 2022-08-22 at 14-31-15" src="https://user-images.githubusercontent.com/2749938/185925093-5ad202c7-10fa-47de-a52f-74e39e83e3c2.png">


* Click on one of the categories or popular searches
* The plugins should load and the top bread crumb should work as logged in
<img width="320" alt="Screenshot on 2022-08-22 at 14-37-03" src="https://user-images.githubusercontent.com/2749938/185925278-a0b0f6e2-a441-49ff-8cdf-17511bc13c27.png">


* The search box should also work and the breadcrumbs should update
<img width="320" alt="Screenshot on 2022-08-22 at 14-35-57" src="https://user-images.githubusercontent.com/2749938/185925323-22fcd4d8-b84c-4c0d-a9d6-e0418294a277.png">


* Clicking on any of the plugins should show the Plugin Details page
<img width="320" alt="Screenshot on 2022-08-22 at 14-37-36" src="https://user-images.githubusercontent.com/2749938/185925399-2b76f86a-6029-4807-bdf0-679c68385065.png">


* Go to `/es/plugins` (or any other supported locale prefix)
* Repeat the steps above
* Make sure the translations work as they do when logged in (NOTE: some plugins are not translated)
* The `/es` locale prefix should be maintained in the URLs while navigating through the pages
<img width="320" alt="Screenshot on 2022-08-22 at 15-52-29" src="https://user-images.githubusercontent.com/2749938/185925868-f9825632-68e2-4106-9089-549f705bcced.png">
<img width="320" alt="Screenshot on 2022-08-22 at 15-54-02" src="https://user-images.githubusercontent.com/2749938/185925956-4ed08c53-a242-41bf-81fe-dfe1c748a3e6.png">


* all the links that should be inaccessible and should redirect to log in
  * `/plugins/manage`
  * `/plugins/upload`
  * `/plugins/setup`
  * `/plugins/active`
  * `/plugins/inactive`
  * `/plugins/updates`
  * `/plugins/wordpress-seo-premium/eligibility`
  * any link which has a `SITE_ID` in it (ex. `/plugins/browse/seo/[SITE_ID]`)


* log in and visit the same links
* everything should work as before

#### TODOs

- [x] Enable Plugins pages for logged-out users (Client-side render + localized)
- [x] Enable Server-side rendering for Plugins pages (#66344)
- [x] Replace popular search words and categories spans with crawlable links
- [x] Implement the Logged-out Plugin details page CTA pointing to the /pricing page
